proxy/proxy/client.go

```diff
@@ -91,6 +95,11 @@ type cacheEntry struct {
 // proxy them to the destination instance. It blocks until connSrc is closed.
 func (c *Client) Run(connSrc <-chan Conn) {
 	for conn := range connSrc {
+		if c.MaxConnections > 0 && len(c.Conns.m) >= c.MaxConnections {
```

Comment:
Great feedback, thanks!
I knew what I implemented was very far from perfect, but was kind of lost where to put the connection limit check.

At first, I wanted to place it in the Accept loop but I did not have access to the number of actual connections there.

I'll look into adding a counter like you suggest.